runtime/stdlib/test.go

```diff
@@ -1033,3 +1374,251 @@ func (e TestFailedError) Unwrap() error {
 func (e TestFailedError) Error() string {
 	return fmt.Sprintf("test failed: %s", e.Err.Error())
 }
+
+// 'Test.DefaultMatcher' struct.
+//
+// This is the default implementation of 'Test.Matcher' interface.
+// It accepts 'Any' for the test function. Hence, can be used with both structs and resources.
+// But the usage is limited by the constructor.
+// i.e: the constructor is hidden from users, to avoid misuses (see below).
+// Instead, use 'Test.NewMatcher()' to construct a matcher that test only 'AnyStruct'.
+//
+// Limitations:
+// Handling resource inside matchers is not supported yet, only exception being the built-in matchers.
```

Comment:
Can `AnyStruct` be used instead then?

Reply:
Yeah we could. The only reason to add `Any` here was to allow some built-in matchers like `equal`, `beNil`, etc. to work with resources. If we are to say that we aren't going to support even those with resources (custom matchers are anyway not supported), then we can just make this `AnyStruct`

Reply:
Maybe we can use references for resources?

Reply:
The problem with references is that equality works differently. It checks whether the reference is equal (i.e: they refer to the same resource), but not whether the two resources referred to are equal.

Reply:
yea, that makes sense

Reply:
updated in e0d14e7026c9e533dcbf95fae917631ce580e9e7 to use `AnyStruct`